server/src/routes/v2/apps.js

```diff
@@ -1,24 +1,55 @@
-const Boom = require('@hapi/boom')
-const Joi = require('@hapi/joi')
-
-const AppModel = require('../../models/v2/out/App')
+const AppModel = require('../../models/v1/out/App')
+const { AppStatus } = require('../../enums')
+const { getApps } = require('../../data')
+const { convertAppsToApiV1Format } = require('../v1/apps/formatting')
+const { filterAppsBySpecificDhis2Version } = require('../../utils/filters')
 
 module.exports = [
     {
         method: 'GET',
         path: '/v2/apps',
         config: {
+            auth: false,
             tags: ['api', 'v2'],
-            response: {
-                schema: Joi.array().items(AppModel.def),
-                failAction: (request, h, err) => {
-                    request.logger.info(err)
-                    throw Boom.internal
+            plugins: {
+                pagination: {
+                    enabled: true,
                 },
             },
         },
-        handler: (request, h) => {
-            throw Boom.notImplemented()
+        handler: async (request, h) => {
+            let channels = ['Stable']
+            if (request.query.channels) {
```

Comment:
Is the idea to accept eg. `channels=Stable,Canary` here, and return the union of those channels? We haven't really how we should send "lists" to the server, but I guess delimited values are fine.

We should however verify/validate the query-objects.
Thing is we need a way to convert string into a list, this can probably be done with Joi, but it's not built in anymore. We could add something like this to our extended-Joi:

```
Joi.extend({
  base: joi.array(),
  name: 'stringArray',
  coerce: (value, state, options) => (value.split ? value.split(',') : value)
})
```

Then we could use that with the custom Joi.filter() and Filter-plugin.

```
  validate: {
                query: Joi.object({
                    channels: Joi.filter(Joi.stringArray().items(Joi.valid('Stable', 'Canary'))).description(
                        'Filter by channel'
                    ),
                    types: Joi.filter(Joi.stringArray().items(Joi.valid(...APPTYPES))).description(
                        'Filter by app types'
                    ),
```

In handler:

```
const channels = request.plugins.queryFilter.getFilter('channels').value
```


The idea when implementing this filter-mechanism was that it could be used transparently" by eg doing `?channels=in:Stable,Canary.` Then you would get the filters applied to your query by doing `filters.applyToQuery(query)`. However, that is not implemented for lists yet (works for simple values though). This query might also be a bit too complicated for that anyway, at least for now.

Anyways, that might be a bit out of scope, but it would be great to get some validation on the query-parameters.

Reply:
> Is the idea to accept eg. channels=Stable,Canary here, and return the union of those channels?
Yep

> Anyways, that might be a bit out of scope, but it would be great to get some validation on the query-parameters.
I'll implement the query validation now and we can defer the `applyToQuery` changes for another PR